tools/lsp/preview.rs

```diff
@@ -219,17 +222,85 @@ struct PreviewState {
 }
 thread_local! {static PREVIEW_STATE: std::cell::RefCell<PreviewState> = Default::default();}
 
+pub fn set_design_mode(sender: crate::ServerNotifier, enable: bool) {
+    let mut cache = CONTENT_CACHE.get_or_init(Default::default).lock().unwrap();
+    cache.design_mode = enable;
+
+    configure_design_mode(enable, &sender);
+    send_notification(
+        &sender,
+        if enable { "Design mode enabled." } else { "Design mode disabled." },
+        Health::Ok,
+    );
+}
+
+fn show_document_request_from_element_callback(
+    url: &str,
+    start_line: u32,
+    start_column: u32,
+    end_line: u32,
+    end_column: u32,
+) -> lsp_types::ShowDocumentParams {
+    use lsp_types::{Position, Range, ShowDocumentParams, Url};
+
+    let start_pos = Position::new(start_line - 1, start_column);
+    let end_pos = Position::new(end_line - 1, end_column);
+    let selection = Some(Range::new(start_pos, end_pos));
+
+    let uri = Url::parse(url).unwrap_or_else(|_| Url::parse("file:///invalid").unwrap());
+
+    ShowDocumentParams { uri, external: Some(false), take_focus: Some(true), selection }
+}
+
+fn configure_design_mode(enabled: bool, sender: &crate::ServerNotifier) {
+    let sender = sender.clone();
+    run_in_ui_thread(Box::pin(async move {
+        PREVIEW_STATE.with(|preview_state| {
+            let preview_state = preview_state.borrow();
+            if let Some(handle) = &preview_state.handle {
+                handle.set_design_mode(enabled);
+
+                handle.on_element_selected(Box::new(
+                    move |url: &str,
+                          start_line: u32,
+                          start_column: u32,
+                          end_line: u32,
+                          end_column: u32| {
+                        let Ok(fut) = sender.send_request::<lsp_types::request::ShowDocument>(
+                            show_document_request_from_element_callback(
+                                url,
+                                start_line,
+                                start_column - 1,
```

Comment:
I wonder why you do the -1 there on the column, and not on the line.  But the show_document_request_from_element_callback does the -1 on the line, but not the column.